hydroserving/core/monitoring/service.py

```diff
@@ -99,6 +100,37 @@ def error_rate_metric_spec(interval, threshold=None):
     return res
 
 
+def custom_model_metric_spec(application, operator, threshold):
+    if not isinstance(application, str):
+        raise TypeError("Invalid app_name: {}".format(application))
+    
+    allowed = {
+        "==": "Eq", 
+        "!=": "NotEq", 
+        ">": "Greater", 
+        "<": "Less", 
+        ">=": "GreaterEq", 
+        "<=": "LessEq", 
+    }
+
+    allowed_flat = list(chain(*allowed.items()))
```

Comment:
I think the code is a little bit bloated. Wouldn't this snippet do exactly the same?
```python
operator_val = allowed.get(operator)
if not operator_val:
    raise Error()
return {..., "kind": operator_val}
```

Reply:
If user already presents mapped value e.g. `Eq` you could check it with `operator in allowed.values()`